### DISCUSSION TITLE
Can this run on Ubuntu?

### DISCUSSION THREAD

**Discussion by astewzxc** [open-edge-platform/anomalib#726] [Q&A] [answered]
Hi, thanks to this great work.
I've run Anomalib on Windows 10 PC, and got good outcome.
However, when I try to run it on Ubuntu, I found the number of model parameters are 24.9M(68.9M on Windows),and got significant lower performance compare to that on Windows.

Ubuntu 20.04
NVIDIA CUDA 11.6.0
PyTorch 1.12.0
TensorRT 8.2.3

**Comment by samet-akcay** [comment]
Hi @astewzxc, can you provide more information to be able to reproduce the issue? For example, your config file, the CLI command you ran etc?

**Comment by astewzxc** [comment]
I use a cloud computing service from [here](https://www.twcc.ai/)
The CLI command I ran are:
    pip install anomalib
    python tools/train.py --model patchcore

but got an error:
ImportError: libGL.so.1: cannot open shared object file: No such file or directory

so I ran:
    pip uninstall opencv-python
    pip install opencv-python-headless

then the error resolved, so I ran
    python tools/train.py --model patchcore

again. It worked, but the model parameters are much less than it ran on Windows.
About the config file, I didn't change anything yet.

**Comment by astewzxc** [comment]
The other thing I forgot to mentioned is that the version of Amomalib I run on Ubuntu is newer than on Windows.
If I ran the older version(version ran on Windows), an error occured:
omegaconf.errors.ConfigAttributeError: Missing key pre_trained
![image](https://user-images.githubusercontent.com/98632061/200614155-79ea4f78-2f43-4cde-80c4-9e088bebf4ce.png)

**Comment by astewzxc** [comment]
The config file of the older version is below:

dataset:
  name: mvtec #options: [mvtec, btech, folder]
  format: mvtec
  path: ./datasets/MVTec
  task: segmentation
  category: grid
  image_size: 224
  train_batch_size: 32
  test_batch_size: 1
  num_workers: 8
  transform_config:
    train: null
    val: null
  create_validation_set: false
  tiling:
    apply: false
    tile_size: null
    stride: null
    remove_border_count: 0
    use_random_tiling: False
    random_tile_count: 16

model:
  name: patchcore
  backbone: wide_resnet50_2
  layers:
    - layer2
    - layer3
  coreset_sampling_ratio: 0.1
  num_neighbors: 9
  weight_file: weights/model.ckpt
  normalization_method: min_max # options: [null, min_max, cdf]

metrics:
  image:
    - F1Score
    - AUROC
  pixel:
    - F1Score
    - AUROC
  threshold:
    image_default: 0
    pixel_default: 0
    adaptive: true

project:
  seed: 0
  path: ./results

logging:
  log_images_to: ["local"] # options: [wandb, tensorboard, local]. Make sure you also set logger with using wandb or tensorboard.
  logger: [] # options: [tensorboard, wandb, csv] or combinations.
  log_graph: false # Logs the model graph to respective logger.

# PL Trainer Args. Don't add extra parameter here.
trainer:
  accelerator: auto # <"cpu", "gpu", "tpu", "ipu", "hpu", "auto">
  accumulate_grad_batches: 1
  amp_backend: native
  auto_lr_find: false
  auto_scale_batch_size: false
  auto_select_gpus: false
  benchmark: false
  check_val_every_n_epoch: 1 # Don't validate before extracting features.
  default_root_dir: null
  detect_anomaly: false
  deterministic: false
  devices: 1
  enable_checkpointing: true
  enable_model_summary: true
  enable_progress_bar: true
  fast_dev_run: false
  gpus: null # Set automatically
  gradient_clip_val: 0
  ipus: null
  limit_predict_batches: 1.0
  limit_test_batches: 1.0
  limit_train_batches: 1.0
  limit_val_batches: 1.0
  log_every_n_steps: 50
  log_gpu_memory: null
  max_epochs: 1
  max_steps: -1
  max_time: null
  min_epochs: null
  min_steps: null
  move_metrics_to_cpu: false
  multiple_trainloader_mode: max_size_cycle
  num_nodes: 1
  num_processes: null
  num_sanity_val_steps: 0
  overfit_batches: 0.0
  plugins: null
  precision: 32
  profiler: null
  reload_dataloaders_every_n_epochs: 0
  replace_sampler_ddp: true
  strategy: null
  sync_batchnorm: false
  tpu_cores: null
  track_grad_norm: -1
  val_check_interval: 1.0 # Don't validate before extracting features.

**Comment by astewzxc** [comment, marked as answer]
I found out it might be the version problem. I ran v0.3.2 on Ubuntu and got good results.

**Comment by samet-akcay** [comment]
Thanks for the update @astewzxc. I'm moving this to discussions now.
